File: Longest Common Prefix - Leetcode 14/Longest Common Prefix - Leetcode 14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        common_prefix = ""
        min_length = min(len(word) for word in strs)
        for i in range(min_length):
            char = strs[0][i]
            if all(word[i] == char for word in strs):
                common_prefix += char
            else:
                break
        return common_prefix

File: Longest Common Prefix - Leetcode 14/Longest Common Prefix - Leetcode 14.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minLen = Integer.MAX_VALUE;
        for (String str : strs){
            int strLen = str.length();
            if(strLen < minLen){
                minLen = strLen;
            }
        }

        int i = 0;
        while(i < minLen){
            char theLetter = strs[0].charAt(i);
            for (String str: strs) {
                if(theLetter != str.charAt(i)){
                    return strs[0].substring(0,i);
                }
            }
            i++;
        }
        return strs[0].substring(0,i);
    }
}